examples/save_to_file/example.exs

```diff
@@ -0,0 +1,206 @@
+
+  @impl true
+  def handle_info({:gun_down, _, :ws, :closed, _}, state) do
+    Logger.info("Server closed ws connection. Exiting")
+    {:stop, :normal, state}
+  end
+
+  @impl true
+  def handle_info(:ws_ping, state) do
+    Process.send_after(self(), :ws_ping, 1000)
+    :gun.ws_send(state.conn, state.stream, :ping)
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_info({:gun_ws, _, _, {:text, msg}}, state) do
+    msg
+    |> Jason.decode!()
+    |> handle_ws_message(state)
+
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_info({:gun_ws, _, _, {:close, code}}, _state) do
+    Logger.info("Signalling connection closed with code: #{code}. Exiting")
+    exit(:ws_down)
+  end
+
+  @impl true
+  def handle_info({:ex_webrtc, _pid, msg}, state) do
+    state = handle_webrtc_message(msg, state)
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_info(msg, state) do
+    Logger.warning("Received unknown msg: #{inspect(msg)}")
+    {:noreply, state}
+  end
+
+  defp handle_ws_message(%{"type" => "offer", "sdp" => sdp}, %{peer_connection: pc} = state) do
+    offer = %SessionDescription{type: :offer, sdp: sdp}
+    Logger.info("Received SDP offer: #{inspect(offer.sdp)}")
+    :ok = PeerConnection.set_remote_description(pc, offer)
+    {:ok, answer} = PeerConnection.create_answer(pc)
+    :ok = PeerConnection.set_local_description(pc, answer)
+    Logger.info("Sent SDP answer: #{inspect(answer.sdp)}")
+    msg = %{"type" => "answer", "sdp" => answer.sdp}
+    :gun.ws_send(state.conn, state.stream, {:text, Jason.encode!(msg)})
+  end
+
+  defp handle_ws_message(%{"type" => "ice", "data" => data}, state) do
+    Logger.info("Received remote ICE candidate: #{inspect(data)}")
+
+    candidate = %IceCandidate{
+      candidate: data["candidate"],
+      sdp_mid: data["sdpMid"],
+      sdp_m_line_index: data["sdpMLineIndex"],
+      username_fragment: data["usernameFragment"]
+    }
+
+    :ok = PeerConnection.add_ice_candidate(state.peer_connection, candidate)
+  end
+
+  defp handle_ws_message(msg, _state) do
+    Logger.info("Received unexpected message: #{inspect(msg)}")
+  end
+
+  defp handle_webrtc_message({:ice_candidate, candidate}, state) do
+    candidate = %{
+      "candidate" => candidate.candidate,
+      "sdpMid" => candidate.sdp_mid,
+      "sdpMLineIndex" => candidate.sdp_m_line_index,
+      "usernameFragment" => candidate.username_fragment
+    }
+
+    msg = %{"type" => "ice", "data" => candidate}
+    :gun.ws_send(state.conn, state.stream, {:text, Jason.encode!(msg)})
+    state
+  end
+
+  defp handle_webrtc_message({:track, _track}, state) do
+    <<fourcc::little-32>> = "VP80"
+
+    # Width, height and FPS (timebase_denum/num)
+    # are the same as we set in video constraints
+    # on the frontend side (in getUserMedia).
+    # However, keep in mind they can change in time
+    # so this is best effort saving.
+    # num_frames is set to 900 but it will be updated 
+    # on calling IVFWriter.close
```

Comment:
There's no call to `IVFWriter.close` anywhere here, I was scratching my head thinking I missed it in the `IFVWriter` PR review, but there seems to be no such function.

Reply:
Yeah, sorry there was such function but I deleted it and restored again